app/src/main/java/ch/epfl/skysync/viewmodel/FinishedFlightsViewModel.kt

```diff
@@ -0,0 +1,135 @@
+package ch.epfl.skysync.viewmodel
+
+import android.util.Log
+import androidx.compose.runtime.Composable
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import androidx.lifecycle.viewmodel.compose.viewModel
+import ch.epfl.skysync.Repository
+import ch.epfl.skysync.components.SnackbarManager
+import ch.epfl.skysync.models.UNSET_ID
+import ch.epfl.skysync.models.flight.FinishedFlight
+import ch.epfl.skysync.models.reports.Report
+import ch.epfl.skysync.models.user.Admin
+import ch.epfl.skysync.models.user.Crew
+import ch.epfl.skysync.models.user.Pilot
+import ch.epfl.skysync.models.user.User
+import ch.epfl.skysync.util.WhileUiSubscribed
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+
+/** ViewModel for the user for the finished flights */
+class FinishedFlightsViewModel(val repository: Repository, val userId: String?) : ViewModel() {
```

Comment:
You should take a non-nullable user id `val userId: String`, at this place in the application we know the uid, this will simplify the logic and remove unnecessary checks

Reply:
Changed it